lib/arjdbc/jdbc/adapter.rb

```diff
@@ -250,12 +250,6 @@ def exec_query_raw(sql, name = 'SQL', binds = [], &block)
         end
       end
 
-      # @private
-      # @override
-      def select_rows(sql, name = nil, binds = [])
```

Comment:
also, I do not understand the removal from the desc or comit message: "Fixed issue with User.exists? 1 call"
could you elaborate? are you trying to fit the base adapter (un-used within core) with the DB2 adapter as is?

Reply:
This was a tricky one.

When I execute query `User.find(1)` it does what you expect, executes this query: 
`SELECT  * FROM USERS WHERE id = ? LIMIT ?  [["id", "1"], ["LIMIT", 1]]`

However just doing `User.exists?(1)` does this:
```
ActiveRecord::StatementInvalid: ActiveRecord::JDBCError: [SQL0313] Number of host variables not valid.: SELECT  1 AS one FROM USER WHERE id = ? LIMIT ?
```

It lost the binds variables (`[["id", "1"], ["LIMIT", 1]]`)

Originally I though, yeah, another DB2 bug. But DB2 does not have `select_rows`. And if you print the binds variables from `select_rows`, it's empty

However the binds do exists. The first variable (`sql`) is changed to `arel` which contains building query information. Then AR52 [select_rows](https://github.com/rails/rails/blob/fc5dd0b85189811062c85520fd70de8389b55aeb/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb#L80) calls  [select_all](https://github.com/rails/rails/blob/fc5dd0b85189811062c85520fd70de8389b55aeb/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb#L46) which parses out the binds from the arel query.

Removing `select_rows` from JDBC fixed the issue

Reply:
ah yes, the AREL 5.2 change where the bind values are in the AST and need to be extracted. Nice catch.